### PR TITLE
MODINREACH-244 INN-Reach Circulation Flow Terminating Transaction Handling: Final Item Check-in (Borrowing Site)

### PR DESCRIPTION
**Pull request by maryna-steshenko**
## Purpose

Remove identifying information from transaction and item records when the FINAL_CHECKIN message is received.
JIRA: [MODINREACH-244](https://issues.folio.org/browse/MODINREACH-244)
## Approach

- Add removing of identifying fields from transaction record.
- Add removing of identifying field from FOLIO item record.
- InnReachCirculationController test updated.

## Pre-Merge Checklist:
Before merging this PR, please go through the following list and take appropriate actions.

- Does this PR meet or exceed the expected quality standards?
  - [ ] Code coverage on new code is 80% or greater
  - [ ] Duplications on new code is 3% or less
  - [ ] There are no major code smells or security issues
- Does this introduce breaking changes?
  - [ ] Were any API paths or methods changed, added or removed?
  - [ ] Were there any schema changes?
  - [ ] Did any of the interface versions change?
  - [ ] Were permissions changed, added, or removed?
  - [ ] Are there new interface dependencies?
  - [ ] There are no breaking changes in this PR.
  
If there are breaking changes, please **STOP** and consider the following:

- What other modules will these changes impact?
- Do JIRAs exist to update the impacted modules?
  - [ ] If not, please create them
  - [ ] Do they contain the appropriate level of detail?  Which endpoints/schemas changed, etc.
  - [ ] Do they have all they appropriate links to blocked/related issues?
- Are the JIRAs under active development?  
  - [ ] If not, contact the project's PO and make sure they're aware of the urgency.
- Do PRs exist for these changes?
  - [ ] If so, have they been approved?

Ideally all of the PRs involved in breaking changes would be merged in the same day to avoid breaking the folio-testing environment.  Communication is paramount if that is to be achieved, especially as the number of intermodule and inter-team dependencies increase.  

While it's helpful for reviewers to help identify potential problems, ensuring that it's safe to merge is ultimately the responsibility of the PR assignee.

### CODE REVIEW

File: src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java [hunk 1]
```diff
@@ -416,9 +417,36 @@ public InnReachResponseDTO finalCheckIn(String trackingId, String centralCode, B
...
 
+  private void removeFinalCheckInFieldsFromItem(InnReachTransaction transaction) {
+    var itemBarcode = transaction.getHold().getFolioItemBarcode();
+    var itemDTO = itemService.findItemByBarcode(itemBarcode).orElseThrow(
+      () -> new EntityNotFoundException("FOLIO item record is not found for barcode: " + itemBarcode));
+
+    itemDTO.setBarcode(null);
+
+    itemService.update(itemDTO);
+  }
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java]: there is already logic implemented for clearing Patron Hold inventory records info. I assume here we should apply the same approach. Clarify with PO if we need to remove a callNumber of item/holding

https://github.com/folio-org/mod-inn-reach/blob/b33d7e9f9966975ec8ad51f59abf57f665e85db8/src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java#L225-L226

Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java]: there is already logic implemented for clearing Patron Hold inventory records info. I assume here we should apply the same approach. Clarify with PO if we need to remove a callNumber of item/holding

https://github.com/folio-org/mod-inn-reach/blob/b33d7e9f9966975ec8ad51f59abf57f665e85db8/src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java#L225-L226

**Review by oliinyko — APPROVED** [review]

File: src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java [hunk 2]
```diff
@@ -1033,6 +1034,8 @@ void checkTransactionIsInStateItemInTransitOrReturnUncirculated(InnReachTransact
 
     assertEquals(HttpStatus.OK, responseEntity.getStatusCode());
     assertEquals(FINAL_CHECKIN, transactionAfter.getState());
+    assertNull(transactionAfter.getHold().getPatronId());
+    assertNull(transactionAfter.getHold().getFolioItemBarcode());
```
Comment (dmtkachenko) [inline src/test/java/org/folio/innreach/controller/d2ir/InnReachCirculationControllerTest.java]: what about rest of the fileds, shoudn;t they be validated also

File: src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java [hunk 3]
```diff
@@ -407,9 +408,27 @@ public InnReachResponseDTO finalCheckIn(String trackingId, String centralCode, B
...
     transaction.setState(FINAL_CHECKIN);
 
+    removeItemTransactionInfo(transaction.getHold().getFolioItemId())
+      .ifPresent(this::removeHoldingsTransactionInfo);
+    removeFinalCheckInFieldsFromHold(transaction);
+
     return success();
   }
 
+  private void removeFinalCheckInFieldsFromHold(InnReachTransaction transaction) {
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/domain/service/impl/CirculationServiceImpl.java]: lets move this method to InnReachTransactionUtils and rename it to something more generic like `clearPatronAndItemInfo`

File: src/main/java/org/folio/innreach/util/InnReachTransactionUtils.java [hunk 4]
```diff
@@ -14,4 +15,17 @@ public static void verifyState(InnReachTransaction transaction, InnReachTransact
     Assert.isTrue(ArrayUtils.contains(expectedStates, actualState), "Unexpected transaction state: " + actualState);
   }
 
+  public static void clearPatronAndItemInfo(InnReachTransaction transaction) {
+    var patronHold = (TransactionPatronHold) transaction.getHold();
```
Comment (oliinyko) [inline src/main/java/org/folio/innreach/util/InnReachTransactionUtils.java]: is there any reason to cast it to TransactionPatronHold?

**Review by dmtkachenko — APPROVED** [review]

**Pull request merged**